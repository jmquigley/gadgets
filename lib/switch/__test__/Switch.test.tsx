'use strict';

import * as assert from 'assert';
import {mount, shallow} from 'enzyme';
import * as React from 'react';
import * as sinon from 'sinon';
import {getDefaultSwitchProps, Switch, SwitchType} from '../index';

test('Test retrieval of the Switch props object', () => {
	const props = getDefaultSwitchProps();

	assert(props);
	expect(props).toMatchSnapshot();
});

test('Test creation of an inny type Switch', () => {
	const ctl = shallow(
		<Switch initialState={true} switchType={SwitchType.inny} />
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test creation of an outy type Switch', () => {
	const ctl = shallow(
		<Switch initialState={true} switchType={SwitchType.outy} />
	);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test disabling of the Switch control', () => {
	const ctl = mount(<Switch disabled />);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test making the Switch control invisible', () => {
	const ctl = mount(<Switch visible={false} />);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
});

test('Test the click handler on the Switch control', () => {
	const click = sinon.spy();
	const ctl = shallow(<Switch onClick={click} />);

	assert(ctl);
	expect(ctl).toMatchSnapshot();
	assert(!ctl.state('toggle'));

	const btn = ctl.find('.ui-switch-button');

	btn.simulate('click');
	assert(click.calledOnce);
	assert(click.calledWith(true));
	assert(ctl.state('toggle'));

	btn.simulate('click');
	assert(click.calledTwice);
	assert(click.calledWith(false));
	assert(!ctl.state('toggle'));
});
