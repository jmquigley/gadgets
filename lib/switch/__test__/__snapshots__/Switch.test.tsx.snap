// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test creation of an inny type Switch 1`] = `
<div
  className="ui-switch container normal"
  style={
    Object {
      "height": "24px",
    }
  }
>
  <div
    className="ui-switch-slider slider normalRect ui-slider-on sliderOn"
  >
    <div
      className="ui-switch-button buttonRightInny ripple"
      onClick={[Function]}
      style={
        Object {
          "height": "12.8px",
          "width": "12.8px",
        }
      }
    />
  </div>
</div>
`;

exports[`Test creation of an outy type Switch 1`] = `
<div
  className="ui-switch container normal"
  style={
    Object {
      "height": "24px",
    }
  }
>
  <div
    className="ui-switch-slider slider normalRect ui-slider-on sliderOn"
  >
    <div
      className="ui-switch-button buttonRightOuty ripple"
      onClick={[Function]}
      style={
        Object {
          "height": "24px",
          "width": "24px",
        }
      }
    />
  </div>
</div>
`;

exports[`Test disabling of the Switch control 1`] = `
<Switch
  className=""
  contentEditable={false}
  direction="right"
  disabled={true}
  focus={false}
  height=""
  id=""
  initialState={false}
  location="none"
  noedit={false}
  nohover={false}
  noripple={false}
  onClick={[Function]}
  selected={false}
  sizing="normal"
  style={Object {}}
  switchType={1}
  testing={false}
  visible={true}
  width=""
>
  <div
    className="ui-switch container normal disabled nohover"
    style={
      Object {
        "height": "24px",
      }
    }
  >
    <div
      className="ui-switch-slider slider normalRect ui-slider-off sliderOff"
    >
      <div
        className="ui-switch-button buttonLeftOuty"
        onClick={[Function]}
        style={
          Object {
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </div>
  </div>
</Switch>
`;

exports[`Test making the Switch control invisible 1`] = `
<Switch
  className=""
  contentEditable={false}
  direction="right"
  disabled={false}
  focus={false}
  height=""
  id=""
  initialState={false}
  location="none"
  noedit={false}
  nohover={false}
  noripple={false}
  onClick={[Function]}
  selected={false}
  sizing="normal"
  style={Object {}}
  switchType={1}
  testing={false}
  visible={false}
  width=""
>
  <div
    className="ui-switch container normal invisible"
    style={
      Object {
        "height": "24px",
      }
    }
  >
    <div
      className="ui-switch-slider slider normalRect ui-slider-off sliderOff"
    >
      <div
        className="ui-switch-button buttonLeftOuty ripple"
        onClick={[Function]}
        style={
          Object {
            "height": "24px",
            "width": "24px",
          }
        }
      />
    </div>
  </div>
</Switch>
`;

exports[`Test retrieval of the Switch props object 1`] = `
Object {
  "className": "",
  "contentEditable": false,
  "direction": "right",
  "disabled": false,
  "focus": false,
  "height": "",
  "id": "",
  "initialState": false,
  "location": "none",
  "noedit": false,
  "nohover": false,
  "noripple": false,
  "onClick": [Function],
  "selected": false,
  "sizing": "normal",
  "style": Object {},
  "switchType": 1,
  "testing": false,
  "visible": true,
  "width": "",
}
`;

exports[`Test the click handler on the Switch control 1`] = `
<div
  className="ui-switch container normal"
  style={
    Object {
      "height": "24px",
    }
  }
>
  <div
    className="ui-switch-slider slider normalRect ui-slider-off sliderOff"
  >
    <div
      className="ui-switch-button buttonLeftOuty ripple"
      onClick={[Function]}
      style={
        Object {
          "height": "24px",
          "width": "24px",
        }
      }
    />
  </div>
</div>
`;
